apps/assessment_registry/serializers.py

```diff
@@ -1,42 +1,134 @@
 from rest_framework import serializers
 
+from utils.common import get_hierarchy_level
 from user_resource.serializers import UserResourceSerializer
-from deep.serializers import ProjectPropertySerializerMixin, TempClientIdMixin
+from deep.serializers import ProjectPropertySerializerMixin, TempClientIdMixin, IntegerIDField
 
 from .models import (
     AssessmentRegistry,
     MethodologyAttribute,
     AdditionalDocument,
+    SummaryIssue,
+    Summary,
+    SummarySubPillarIssue,
+    SummaryFocus,
+    SummarySubDimmensionIssue,
     ScoreRating,
     ScoreAnalyticalDensity,
+    Answer,
 )
 
 
 class MethodologyAttributeSerializer(TempClientIdMixin, serializers.ModelSerializer):
+    id = IntegerIDField(required=False)
+
     class Meta:
         model = MethodologyAttribute
         fields = (
-            "client_id", "data_collection_technique", "sampling_approach", "sampling_size",
+            "id", "client_id", "data_collection_technique", "sampling_approach", "sampling_size",
             "proximity", "unit_of_analysis", "unit_of_reporting",
         )
 
 
 class AdditionalDocumentSerializer(TempClientIdMixin, UserResourceSerializer):
+    id = IntegerIDField(required=False)
+
     class Meta:
         model = AdditionalDocument
-        fields = ("client_id", "document_type", "file", "external_link",)
+        fields = ("id", "client_id", "document_type", "file", "external_link",)
 
 
-class ScoreRatingSerializer(TempClientIdMixin, UserResourceSerializer):
+class IssueSerializer(UserResourceSerializer):
+    class Meta:
+        model = SummaryIssue
+        fields = (
+            'sub_pillar', 'sub_dimmension', 'parent', 'label'
+        )
+
+    def validate(self, data):
+        if data.get('sub_pillar') is not None and data.get('sub_dimmension') is not None:
+            raise serializers.ValidationError("Cannot select both sub_pillar and sub_dimmension field.")
+        if not data.get('sub_pillar') and not data.get('sub_dimmension'):
+            raise serializers.ValidationError("Either sub_pillar or sub_dimmension must be selected")
+        if data.get('parent') is not None:
+            if data.get('sub_pillar') is not None:
+                if data.get('sub_pillar') != data.get('parent').sub_pillar:
+                    raise serializers.ValidationError("sub_pillar does not match between parent and child.")
+
+            if data.get('sub_dimmension') is not None:
+                if data.get('sub_dimmension') != data.get('parent').sub_dimmension:
+                    raise serializers.ValidationError("sub_dimmension does not match between child and parent.")
+        if data.get('parent'):
+            hierarchy_level = get_hierarchy_level(data.get('parent'))
+            if hierarchy_level > 2:
+                raise serializers.ValidationError("Cannot create issue more than two level of hierarchy")
+
+        return data
```

Comment:
After using non-zero values for enums

```suggestion
    def validate(self, data):

        sub_pillar = data.get('sub_pillar')
        sub_dimmension = data.get('sub_dimmension')
        parent = data.get('parent')

        if all([sub_pillar, sub_dimmension])
            raise serializers.ValidationError("Cannot select both sub_pillar and sub_dimmension field.")

        if not any([sub_pillar, sub_dimmension])
            raise serializers.ValidationError("Either sub_pillar or sub_dimmension must be selected")

        if parent:
            if sub_pillar and sub_pillar != parent.sub_pillar:
                raise serializers.ValidationError("sub_pillar does not match between parent and child.")

            if sub_dimmension and sub_dimmension != parent.sub_dimmension:
                raise serializers.ValidationError("sub_dimmension does not match between child and parent.")

            hierarchy_level = get_hierarchy_level(parent)
            if hierarchy_level > 2:
                raise serializers.ValidationError("Cannot create issue more than two level of hierarchy")

        return data
```